### PR TITLE
Update pre-commit hooks

### PR DESCRIPTION
**Pull request by bdice**
## Description
This PR updates rmm's pre-commit hooks and CI style check scripts.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/rmm/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by harrism** [comment]
Burn down starts tomorrow, so if you want to get part of this in 22.12 suggest opening the PRs now.

**Comment by bdice** [comment]
@harrism It's not as time sensitive as GitHub Actions for cuDF, which is my primary project at the moment. Since this is partially blocked by #1151 anyway, I am going to de-prioritize this and come back to it at a convenient time (probably just after burndown, targeting 23.02). Since it's mostly for developer convenience, the release timing is not crucial.

**Review by harrism — APPROVED** [review]

**Review by raydouglass — APPROVED** [review]

File: ci/check_style.sh [hunk 1]
```diff
@@ -16,124 +16,10 @@ conda activate checks
 
 set +e
```
Comment (ajschmidt8) [inline ci/check_style.sh:17]:
```suggestion
```

now that we've removed the explicit `exit` command at the bottom of this script, this `set +e` should be removed also.

**Review by ajschmidt8 — CHANGES_REQUESTED** [review]

**Review by harrism — APPROVED** [review]

**Review by ajschmidt8 — APPROVED** [review]

**Comment by bdice** [comment]
@gpucibot merge

**Pull request merged**